dimagi_lake/aggregation/aggregation_helpers/base_helper.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC
+
+from spark_session_handler import SPARK
+
+
+class BaseAggregationHelper(ABC):
+    """
+    Abstract Class defining the template for
+    aggregation Helper class containing the aggregation logic.
+    Each Aggregation helper is connected to different Table Schema class in aggregation/agg_table_schema
+    with a class attribute `_aggregator`
+    """
+    def __init__(self, database_name, domain, month, schema):
+        self.database_name = database_name
+        self.domain = domain
+        self.month = month
+        self.schema = schema
+
+    def aggregate(self):
```

Comment:
I'm kind of confused by the interface to this class. If `aggregate` is meant to be customized for each table, then it seems like the internal functions it calls (`agg_base_data`, `rollup_data`, etc.) would be internal methods managed by the subclasses. Would those sub-functions ever be called anywhere besides `aggregate()`?

Reply:
no,  I think those internal methods are only called through aggregate function only